events/events.go

```diff
@@ -31,9 +34,100 @@ type EventListener interface {
 	HandleEvent(event Event)
 }
 
+// EventEmitter describes an interface that can be used to emit events
+type EventEmitter interface {
```

Comment:
Is it possible to combine EventEmitter and EventRegistrar? I was a bit confused why we have to; since having two seems to suggest that they should interact. I think having one EventEmitter for the three functions would be nice. What do you think?